src/plot_api/plot_api.js

```diff
@@ -2032,12 +2095,16 @@ function _relayout(gd, aobj) {
         }
     }
 
+    // now we've collected component edits - execute them all together
+    for(arrayStr in arrayEdits) {
+        var finished = manageArrays.editContainerArray(gd,
+            Lib.nestedProperty(layout, arrayStr), arrayEdits[arrayStr], flags);
```

Comment:
Same feedback. If this is the only place it's used, perhaps the supplydefaults and draw parts should be outside the function that manages arrays? You have a better sense than I do though, so feel free to disregard :)